terraform/node_resource_refresh.go

```diff
@@ -176,3 +172,77 @@ func (n *NodeRefreshableManagedResourceInstance) evalTreeManagedResource() EvalN
 		},
 	}
 }
+
+func (n *NodeRefreshableManagedResourceInstance) evalTreeManagedScaleOutResource() EvalNode {
```

Comment:
I think without the context of this PR the name of this method will be a little confusing/unclear. Do you think there's a reasonable comment we could put here to explain what situation this function is addressing? Might be good to also summarize the same idea at the `n.ResourceState == nil` branch above, too.

Maybe it would be easier to follow through this all if this method were called `evalTreeManagedResourceNoState`, since it seems like in any case this would run for _any_ resource that currently has no state.